camel/src/main/java/com/github/theprez/manzan/routes/event/WatchMsgEventSockets.java

```diff
@@ -32,16 +30,22 @@ public void configure() {
             .unmarshal().json(JsonLibrary.Jackson, Map.class)
             .routeId("manzan_msg:"+m_name)
             .setHeader(EVENT_TYPE, constant(ManzanEventType.WATCH_MSG))
-            .setHeader("session_id", simple("${body[sessionId]}"))
+            .setHeader("session_id", simple("${body[SESSION_ID]}"))
             .setHeader("data_map", simple("${body}"))
             .marshal().json(true) //TODO: skip this if we are applying a format
             .setBody(simple("${body}\n"))
             .process(exchange -> {
-                if (null != m_formatter) {
+                String sessionId = exchange.getIn().getHeader("session_id", String.class);
+                String format = m_formatMap.get(sessionId);
+                if (format != null) {
+                    ManzanMessageFormatter m_formatter = new ManzanMessageFormatter(format);
                     exchange.getIn().setBody(m_formatter.format(getDataMap(exchange)));
+                    String destinations = m_destMap.get(sessionId); // Get destinations from m_destMap
+                    exchange.getIn().setHeader("destinations", destinations);
                 }
             })
-            .recipientList(constant(getRecipientList())).parallelProcessing().stopOnException().end();
+                .recipientList(header("destinations"))
+                .parallelProcessing().stopOnException();
```

Comment:
Is the previous `.end()` needed?

Reply:
Maybe.. I've added it back, just in case